src/main/java/org/shanerx/tradeshop/data/config/Setting.java

```diff
@@ -223,14 +306,40 @@ public String getPath() {
         return path;
     }
 
+    private String processMapValue(Map<?, ?> valueMap, String localLead) {
+        StringBuilder processed = new StringBuilder();
+
+        List<Object> sortedKeys = new ArrayList<>(valueMap.keySet());
+        Collections.sort(sortedKeys, (o1, o2) -> o1.toString().compareTo(o1.toString()));
+
+        //PLUGIN.getDebugger().log("Sorted Value: " + sortedValue +"\nValue: " + valueMap, DebugLevels.DATA_ERROR);
+
+        for (Object key : sortedKeys) {
+            Object value = valueMap.get(key);
+            if (value instanceof Map) {
+                processed.append(localLead).append(key.toString()).append(":\n");
+                processed.append(processMapValue(((Map<?, ?>) value), localLead + leadIncrease));
+            } else {
+                processed.append(localLead).append(key.toString()).append(": ").append(new Yaml().dump(value));
+            }
+        }
+
+        return processed.toString();
+    }
+
     public String getFileString() {
         StringBuilder keyOutput = new StringBuilder();
 
         if (!getPreComment().isEmpty()) {
             keyOutput.append(section.getSectionLead()).append("# ").append(PLUGIN.getSettingManager().fixCommentNewLines(section.getSectionLead(), getPreComment())).append("\n");
         }
 
-        keyOutput.append(section.getSectionLead()).append(getKey()).append(": ").append(new Yaml().dump(getSetting()));
+        if (defaultValue instanceof Map) {
+            keyOutput.append(section.getSectionLead()).append(getKey()).append(":\n");
+            keyOutput.append(processMapValue(((Map<?, ?>) defaultValue), section.getSectionLead() + leadIncrease));
+        } else {
+            keyOutput.append(section.getSectionLead()).append(getKey()).append(": ").append(new Yaml().dump(getSetting()));
```

Comment:
same here, to hell with performance enhancements from StringBuilder, I think String.format makes it clearer